cobigen-angular-client-generation/index.asciidoc

```diff
@@ -0,0 +1,94 @@
+buildNg("cobigenangularexample")
+runClientNg("cobigenangularexample/angular", { "startupTime": 200, "port": 4200, "path": "" })
+createFile("cobigenangularexample/core/src/main/java/com/example/application/cobigenangularexample/customermanagement/dataaccess/api/CustomerEntity.java", "files/CustomerEntity.java")
+adaptTemplatesCobiGen()
+cobiGenJava("cobigenangularexample/core/src/main/java/com/example/application/cobigenangularexample/customermanagement/dataaccess/api/CustomerEntity.java",[1,3,5,6,8])
+--
+The CobiGen code generator will generate some java classes for you. These contain code for basic CRUD operations, REST service handling and data access.
+
+For example, the following files are generated by CobiGen when using the specified templates:
+
+(1) CRUD logic: Generates the logic layer and implementations for some use cases.
+- `devonfw/workspaces/main/cobigenangularexample/core/src/main/java/com/example/application/cobigenangularexample/customermanagement/logic/impl/CustomermanagementImpl.java`{{open}}
+- `devonfw/workspaces/main/cobigenangularexample/core/src/main/java/com/example/application/cobigenangularexample/customermanagement/logic/impl/usecase/UcManageCustomerImpl.java`{{open}}
+- `devonfw/workspaces/main/cobigenangularexample/core/src/main/java/com/example/application/cobigenangularexample/customermanagement/logic/impl/usecase/UcFindCustomerImpl.java`{{open}}
+- `devonfw/workspaces/main/cobigenangularexample/api/src/main/java/com/example/application/cobigenangularexample/customermanagement/logic/api/Customermanagement.java`{{open}}
+
+(3) CRUD REST services: Generates the service layer with CRUD operations for using in REST services.
+- `devonfw/workspaces/main/cobigenangularexample/api/src/main/java/com/example/application/cobigenangularexample/customermanagement/service/api/rest/CustomermanagementRestService.java`{{open}}
+- `devonfw/workspaces/main/cobigenangularexample/core/src/main/java/com/example/application/cobigenangularexample/customermanagement/service/impl/rest/CustomermanagementRestServiceImpl.java`{{open}}
+
+(5) TO's: Generates the related Transfer Objects.
+- `devonfw/workspaces/main/cobigenangularexample/api/src/main/java/com/example/application/cobigenangularexample/customermanagement/logic/api/to/CustomerEto.java`{{open}}
+- `devonfw/workspaces/main/cobigenangularexample/api/src/main/java/com/example/application/cobigenangularexample/customermanagement/logic/api/to/CustomerSearchCriteriaTo.java`{{open}}
+
+(6) Entity infrastructure: Creates the entity main interface and edits (by a merge) the current entity to extend the newly generated classes.
+- `devonfw/workspaces/main/cobigenangularexample/api/src/main/java/com/example/application/cobigenangularexample/customermanagement/common/api/Customer.java`{{open}}
+- `devonfw/workspaces/main/cobigenangularexample/core/src/main/java/com/example/application/cobigenangularexample/customermanagement/dataaccess/api/CustomerEntity.java`{{open}} (changed)
```

Comment:
Do not use Katacoda Scenario Syntax as this may result in parsing issues. Please use the [tutorial-compiler functions](https://github.com/devonfw-tutorials/tutorial-compiler/wiki/Functions) instead.

Reply:
I think we have also used this block of text in the cobigen tutorial before.